gallifreyan.js

```diff
@@ -209,7 +229,15 @@ function doClick(e) {
             }
         }
     }
-    if (selectedLine != null) selectedCircle = null;    //if we've selected a line, let's unselect a circle
+    if (selectedLine != null) {
+        selectedCircle = null;    //if we've selected a line, let's unselect a circle
+        if (deleteLineMode) {
+            deleteLine(selectedLine);
+            selectedLine = null;
+        }
+    }
+    deleteLineMode = false;

```

Comment:
Okay. Hovewer, I still think this could be put separately:

		if (deleteLineMode) {
			deleteLineMode = false;
			redraw();
		}
	};

This makes it a bit more clear and avoids unnecessary redraws if we're just randomly clicking on a canvas.